src/main/java/frc/robot/subsystems/shooter/Shooter.java

```diff
@@ -0,0 +1,151 @@
+        // https://file.tavsys.net/control/controls-engineering-in-frc.pdf Page 76
+        Vector<N1> A = VecBuilder.fill(-Math.pow(GEAR_RATIO, 2) * Kt / (Kv * OMEGA * J)); //Change the amount of cells and rows
+        Vector<N1> B = VecBuilder.fill(GEAR_RATIO * Kt / (OMEGA * J));
+        LinearSystem<N1, N1, N1> stateSpace = new LinearSystem<>(A, B, Matrix.eye(Nat.N1()), new Matrix<>(Nat.N1(), Nat.N1()));
+        KalmanFilter<N1, N1, N1> kalman = new KalmanFilter<>(Nat.N1(), Nat.N1(), stateSpace,
+                VecBuilder.fill(MODEL_TOLERANCE),
+                VecBuilder.fill(ENCODER_TOLERANCE),
+                Constants.LOOP_PERIOD
+        );
+        LinearQuadraticRegulator<N1, N1, N1> lqr = new LinearQuadraticRegulator<>(stateSpace, VecBuilder.fill(VELOCITY_TOLERANCE),
+                VecBuilder.fill(Constants.NOMINAL_VOLTAGE),
+                Constants.LOOP_PERIOD // time between loops, DON'T CHANGE
+        );
+
+        return new LinearSystemLoop<>(stateSpace, lqr, kalman, Constants.NOMINAL_VOLTAGE, Constants.LOOP_PERIOD);
+    }
+
+    private InputStreamReader readCSV() {
+        InputStream is = getClass().getResourceAsStream(PATH_TO_CSV);
+        assert is != null : "Can't create input stream";
+
+        return new InputStreamReader(is);
+    }
+
+    /**
+     * @return the velocity of the shooter. [RPS]
+     * @see #setVelocity(double)
+     */
+    public double getVelocity() {
+        return unitModel.toVelocity(main.getSelectedSensorVelocity());
+    }
+
+    /**
+     * Set the velocity to apply by the motor.
+     *
+     * @param velocity the desired velocity at which the motor will rotate. [RPS]
+     * @see #setPower(double)
+     */
+    public void setVelocity(double velocity) {
+        stateSpacePredictor.setNextR(VecBuilder.fill(velocity)); //r = reference
+        stateSpacePredictor.correct(VecBuilder.fill(getVelocity()));
+        stateSpacePredictor.predict(Constants.LOOP_PERIOD); //every 20 ms
+
+        double voltageToApply = stateSpacePredictor.getU(0); // u = input, calculated by the input.
+        // returns the voltage to apply (between 0 and 12)
+        setPower(voltageToApply / Constants.NOMINAL_VOLTAGE); // map to be between 0 and 1
+    }
+
+    /**
+     * Set the power to apply by the motor.
+     *
+     * @param power the power at which the motor will rotate. [percentage, between 0 and 1]
+     * @see #stop()
+     */
+    public void setPower(double power) {
+        main.set(TalonFXControlMode.PercentOutput, power, DemandType.ArbitraryFeedForward, Constants.Shooter.ARBITRARY_FEED_FORWARD);
+    }
+
+    /**
+     * Estimate the velocity that the shooter should apply in order to reach the target.
+     *
+     * @param distance the distance from the target.
+     * @return the velocity that should be applied by the shooter in order to reach the target.[RPS]
+     */
+    public double estimateVelocity(double distance) {
```

Comment:
Could you please the noun from "estimate" to something else? I get confused with all the Kalman estimator stuff

Reply:
approximate?

Reply:
I think `calculateVelocityByDistance` or something similar will be the clearest name.

Reply:
I don't think we need to make it more explanatory, because: 
1) it's pretty clear by the parameter that we want to get the distance,
2) we are not intending to share this code with other teams, and we know what is the purpose of this function, AND
3) even if in three years we look into this function, I think we still can to recognize the purpose.

(BTW, you still didn't notice that I don't have units in the distance parameter...🙃, and yes, I will add it soon)

Reply:
> we are not intending to share this code with other teams, and we know what is the purpose of this function

We do when the season ends.
> even if in three years we look into this function, I think we still can to recognize the purpose

[Yes We Can](https://en.wikipedia.org/wiki/Barack_Obama_2008_presidential_campaign#Slogan), but it will be less straight forward (especially with all the Kalman estimator terms flying around).

Reply:
> We do when the season ends.

This repo is already public, and we don't intend to explain them how it works...

> Yes We Can, but it will be less straight forward (especially with all the Kalman estimator terms flying around).

How, When we look into the signature of this function we understand the intention of this function. (btw, You know that I have already changed the name of this function to `evaluateVelocity()`)

Reply:
> This repo is already public

I agree it should be private. However, right now people (probably) aren't looking at our code, but they will when will share it on Chief Delphi.
> When we look into the signature of this function we understand the intention of this function

True, but it's not clear if you look at the name only.
> You know that I have already changed the name of this function to `evaluateVelocity()`

Yes, but I think it's still not perfect.

Reply:
I'm tired of arguing over this nonsense, sure I will change the name